### DISCUSSION TITLE
Python Task Runners

### DISCUSSION THREAD

**Discussion by jwfuller** [nsidc/tech-radar#8] [Languages and Frameworks]
- Invoke
- Fabric

**Comment by trey-stafford** [comment]
Fabric v1 was more common on older projects and we have worked to update such projects to use `invoke` when convenient .

It looks like Fabric v2 was released in January of 2023 and focuses on executing commands on remote hosts. From [https://www.fabfile.org/](https://www.fabfile.org/):

> Fabric is a high level Python (2.7, 3.4+) library designed to execute shell commands remotely over SSH, yielding useful Python objects in return.

**Reply by MattF-NSIDC** [reply]
>It builds on top of [Invoke](https://pyinvoke.org/) (subprocess command execution and command-line features) and [Paramiko](https://paramiko.org/) (SSH protocol implementation), extending their APIs to complement one another and provide additional functionality.

The history of invoke and fabric just got weird :laughing:
